### PR TITLE
PB-318: refactor time slider to composition API

### PR DESCRIPTION
**Pull request by ltkum**
since we are working on the time Slider implementation, we might as refactor it with the composition API

[Test link](https://sys-map.dev.bgdi.ch/preview/compose-api-timeslider/index.html)

### CODE REVIEW

File: src/modules/map/components/toolbox/TimeSlider.vue [hunk 1]
```diff
@@ -1,3 +1,305 @@
...
+}
+
+// dynamic internal data
+const sliderWidth = ref(0)
+const allYears = ref(ALL_YEARS)
+const currentYear = ref(YOUNGEST_YEAR)
+const cursorX = ref(0)
+const playYearsWithData = ref(false)
+const yearCursorIsGrabbed = ref(false)
+const playYearInterval = ref(null)
```
Comment (pakb) [inline src/modules/map/components/toolbox/TimeSlider.vue:63]: I don't think you should store that in a ref, a simple `const` or `let` would do the trick

File: src/modules/map/components/toolbox/TimeSlider.vue [hunk 2]
```diff
@@ -1,3 +1,305 @@
...
+        const [onlyLayerLeft] = newLayers
+        if (!onlyLayerLeft.timeConfig.years.includes(currentYear.value)) {
+            setCurrentYearAndDispatchToStore(onlyLayerLeft.timeConfig.years[0])
+        }
+    }
+})
+
+// we can't watch currentYear and dispatch changes to the store here, otherwise the store gets
+// dispatch too many times when the user is moving the time slider (we wait for mouseup our
+// touchend to commit the change)
```
Comment (pakb) [inline src/modules/map/components/toolbox/TimeSlider.vue:163]: That's good enough, but now that we have some "experience" and some utils to debounce actions, you might want to plan switching to a debounce based store dispatch approach.

Relying on both mouseup and touchend can be a bit tricky on some devices that fire both

File: src/modules/map/components/toolbox/TimeSlider.vue [hunk 3]
```diff
@@ -1,3 +1,305 @@
...
+}
+
+function setSliderWidth() {
+    // 19px of padding (7.5 on both side of the container with p-2 class and 4 with px-1)
+
+    sliderWidth.value = sliderContainer.value.clientWidth - 19 - PLAY_BUTTON_SIZE
+}
+
+function positionNodeLabel(year) {
+    const timestampIndex = ALL_YEARS.indexOf(year) || 1
```
Comment (pakb) [inline src/modules/map/components/toolbox/TimeSlider.vue:208]: if you see this kinds of "default assign" through `||` you can translate them into newer JS with `??` operator instead

**Review by pakb — APPROVED** [review]

File: src/modules/map/components/toolbox/TimeSlider.vue (hunk 2 above)
Comment (ltkum) [reply src/modules/map/components/toolbox/TimeSlider.vue:163]: My goal was mostly to make a 1 to 1 translation. I'll keep that in mind when I'm developing the new features, especially with the datas that are distinct.

**Pull request merged**